--- a/etc/samba/smb.conf.template	2020-03-13 14:56:40.275184245 +0800
+++ b/etc/samba/smb.conf.template	2020-03-13 14:58:38.665344696 +0800
@@ -20,6 +20,8 @@
 	## This parameter controls whether a remote client is allowed or required to use SMB encryption.
 	## It has different effects depending on whether the connection uses SMB1 or SMB2 and newer:
     ## If the connection uses SMB1, then this option controls the use of a Samba-specific extension to the SMB protocol introduced in Samba 3.2 that makes use of the Unix extensions.
+	server min protocol = NT1
+	client min protocol = NT1
 	## If the connection uses SMB2 or newer, then this option controls the use of the SMB-level encryption that is supported in SMB version 3.0 and above and available in Windows 8 and newer. 
 	## (default/auto,desired,required,off)
 	#smb encrypt = default
