From d65ef1d0ddf9a0e7e7d6fb9d4c9d3319d2b311f1 Mon Sep 17 00:00:00 2001
From: W_Y_CPP <383152993@qq.com>
Date: Sun, 28 May 2023 13:44:07 +0900
Subject: [PATCH] fix build

---
 wpa_supplicant/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/wpa_supplicant/Makefile b/wpa_supplicant/Makefile
index ef36b56..55f0c76 100644
--- a/wpa_supplicant/Makefile
+++ b/wpa_supplicant/Makefile
@@ -339,6 +339,7 @@ endif
 
 ifdef CONFIG_MBO
 CONFIG_WNM=y
+NEED_GAS=y
 endif
 
 ifdef CONFIG_WNM
--
2.17.1
